src/renderer/services/tab.ts

```diff
@@ -115,12 +127,99 @@ export const insertAtCursor = (tab: Tab, s: string): Tab => {
     const newTab = removeAtCursor(tab)
     const [newContent, newCursor] = insertAt(newTab.content, s, newTab.cursor)
 
-    return setCursor({ ...tab, content: newContent }, newCursor, false)
+    const newTab1 = { ...tab, content: newContent }
+    newTab1.diffs.done.push({
+        op: 'add',
+        at: newTab.cursor,
+        value: s.split('\n'),
+    })
+    newTab1.diffs.undone = []
```

Comment:
Should implement a function for adding diffs so we won't forget clearing the `undone` stack.